dagshub/upload/wrapper.py

```diff
@@ -202,6 +204,10 @@ def __init__(
 
         self._api = RepoAPI(f"{owner}/{name}", host=self.host, auth=self.auth)
 
+        # For mirror uploading: store the last revision for which we uploaded
+        # When the last revision changes, that means the sync has been complete and we can upload a new batch
+        self._last_upload_revision: Optional[str] = None
+
```

Comment:
Putting this as a variable on the `Repo` object feels wrong - it will cause shenanigans for any user trying to use `Repo.upload_files` directly. It should be a variable in whichever for loop is splitting the upload into chunks.

Reply:
But this way the upload works across the whole repo object.
If the user is, say, training stuff and uploading model results in a loop (new artifact after each loop step), you can't guarantee sync consistency there

Reply:
This is doing exactly the opposite of causing shennanigans.
If I as a user that doesn't upload the whole folder at once, do:
```
repo = Repo("user", "mirror-repo")
repo.upload("file1.txt")
repo.upload("file2.txt")
```
We will break

Reply:
Hmm I guess since the branch is part of the `Repo` object and I can't upload to different branches in parallel, you're probably right...
This makes me think of a different issue - if new_branch is passed, then no check needs to be made

Reply:
Good catch, thank you! I'll fix it

Reply:
I guess it might be a good idea to put a warning that this class is overall not thread safe.